test/e2e/framework/kubernetes/check-pod-status.go

```diff
@@ -85,3 +78,22 @@ func WaitForPodReady(ctx context.Context, clientset *kubernetes.Clientset, names
 	}
 	return nil
 }
+
+func CheckPodRestarts(ctx context.Context, clientset *kubernetes.Clientset, namespace, labelSelector string) error {
+	var podList *corev1.PodList
+	podList, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return fmt.Errorf("error listing Pods: %w", err)
+	}
+
+	for _, pod := range podList.Items {
+		for istatus := range pod.Status.ContainerStatuses {
+			status := &pod.Status.ContainerStatuses[istatus]
+			if status.RestartCount > 0 {
+				return fmt.Errorf("pod %s has %d restarts: status: %+v: %w", pod.Name, status.RestartCount, status, ErrPodCrashed)
+			}
+		}
+	}
+	log.Printf("no restarts detected for pods in namespace \"%s\" with label \"%s\"\n", namespace, labelSelector)
```

Comment:
this could be pretty noisy with a lot of pods under that label, maybe omit and let caller log if error